packages/studio/src/components/NumberPropInput.tsx

```diff
@@ -0,0 +1,55 @@
+import { PropValueType, TypeGuards } from "@yext/studio-plugin";
+import { ChangeEvent, useCallback } from "react";
+import FieldPickerInput from "./FieldPicker/FieldPickerInput";
+
+import TemplateExpressionFormatter from "../utils/TemplateExpressionFormatter";
+
+interface PropInputProps {
+  value: number;
+  onChange: (value: number | string) => void;
+  disabled?: boolean;
+}
+
+/**
+ * Renders the input element of a number prop, that will update the
+ * corresponding prop's value for the active component based on a user's input.
+ */
+export default function NumberPropInput({
+  value,
+  onChange,
+  disabled,
+}: PropInputProps): JSX.Element {
+  const handleChangeEvent = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      const val = Number(e.target.value);
+      onChange(val);
+    },
+    [onChange]
+  );
+
+  const appendField = useCallback(
+    (fieldId: string) => {
+      const documentUsage = "${" + fieldId + "}";
```

Comment:
This syntax only exists to support template string interpolation, and shouldn't be here for numbers

Reply:
Oh? I was using it to help handle the expressions since we want it to accept stuff like document.businessId.

Reply:
you can just have it be `document.businessId` instead of `${document.businessId}`
when you save to file with the template string syntax what does it look like?

Reply:
oh ur right! Ohhh I get why its string thing now